### PR TITLE
feat(azure-iothub): Support User Assigned Managed Identities for import/export jobs

### PR DESCRIPTION
**Pull request by vishnureddy17**

### CODE REVIEW

File: service/src/registry.ts [hunk 1]
```diff
@@ -448,16 +453,28 @@ export class Registry {
...
       const httpHeaders = {
         'Content-Type': 'application/json; charset=utf-8'
       };
       const importRequest = {
         'type': 'import',
         'inputBlobContainerUri': inputBlobContainerUri,
         'outputBlobContainerUri': outputBlobContainerUri,
-        'storageAuthenticationType': 'IdentityBased'
+        'storageAuthenticationType': 'IdentityBased',
+        ...(typeof doneOrIdentity === 'string' && {
```
Comment (anthonyvercolano) [inline service/src/registry.ts:473]: Well, I'll let you explain this to me.

Comment (anthonyvercolano) [reply service/src/registry.ts:473]: Where did you find this?

**Review by anthonyvercolano — APPROVED** [review]

**Review by anthonyvercolano — APPROVED** [review]

File: service/src/registry.ts (hunk 1 above)
Comment (vishnureddy17) [reply service/src/registry.ts:473]: https://amberley.dev/blog/2020-09-07-conditionally-add-to-array-or-obj/#conditionally-add-a-property-to-an-object

**Pull request merged**